Sources/FluidAudio/TextToSpeech/TtsModels.swift

```diff
@@ -0,0 +1,206 @@
+import CoreML
+import Foundation
+import OSLog
+
+@available(macOS 13.0, *)
+public struct TtsModels {
+    private let kokoroModels: [ModelNames.TTS.Variant: MLModel]
+
+    private static let logger = AppLogger(subsystem: "com.fluidaudio.tts", category: "TtsModels")
+
+    public init(models: [ModelNames.TTS.Variant: MLModel]) {
+        self.kokoroModels = models
+    }
+
+    internal var modelsByVariant: [ModelNames.TTS.Variant: MLModel] {
+        kokoroModels
+    }
+
+    public var availableVariants: Set<ModelNames.TTS.Variant> {
+        Set(kokoroModels.keys)
+    }
+
+    public func model(for variant: ModelNames.TTS.Variant = ModelNames.TTS.defaultVariant) -> MLModel? {
+        kokoroModels[variant]
+    }
+
+    public static func download(
+        variants requestedVariants: Set<ModelNames.TTS.Variant>? = nil,
+        from repo: String = TtsConstants.defaultRepository,
+        progressHandler: DownloadUtils.ProgressHandler? = nil
+    ) async throws -> TtsModels {
+        let cacheDirectory = try getCacheDirectory()
+        // Pass Models subdirectory so models end up in ~/.cache/fluidaudio/Models/kokoro/
+        let modelsDirectory = cacheDirectory.appendingPathComponent(TtsConstants.defaultModelsSubdirectory)
+        let targetVariants: [ModelNames.TTS.Variant] = {
+            if let requested = requestedVariants, !requested.isEmpty {
+                return requested.sorted { $0.fileName < $1.fileName }
+            }
+            return ModelNames.TTS.Variant.allCases
+        }()
+        let modelNames = targetVariants.map { $0.fileName }
+        let dict = try await DownloadUtils.loadModels(
+            .kokoro,
+            modelNames: modelNames,
+            directory: modelsDirectory,
+            // Only a small fraction of the model can run on ANE, and compile time takes a long time because of the complicated arch
+            computeUnits: .cpuAndGPU
+        )
```

Comment:
For other reviewers, @Alex-Wengg tried a bunch of different combinations of the base model - some operations are able to run on ANE but the overhead was much greater than the efficiency gains. 

At this point, the main benefit of the CoreML models over MLX are in terms of memory and minor speedups